Compilation problem with rust 1.15.0 nightly?

Hi there,

I'm trying to use _stainless_ in a toy project. However, i'm getting some unexpected compilation errors (shown below).

This is my environnement:

```
rustc --version                                                                                                                                                    
rustc 1.15.0-nightly (3bf2be9ce 2016-11-22)
```

```
cargo --version                                                                                                                                                   
cargo 0.16.0-nightly (d1bfc01 2016-11-18)
```

I'm using the latest stainless version available on [crates.io](https://crates.io/crates/stainless/0.1.9)
```
[dev-dependencies]
stainless = "0.1.9"
```

Just by doing a
```
cargo test                                                                                                                                                         
```

I get this error stack:

```
Compiling stainless v0.1.9
error[E0412]: type name `token::InternedString` is undefined or not in scope
  --> /Users/myuser/.cargo/registry/src/github.com-1ecc6299db9ec823/stainless-0.1.9/src/test.rs:23:30
   |
23 |     pub failing_msg: Option<(token::InternedString, ast::StrStyle)>,
   |                              ^^^^^^^^^^^^^^^^^^^^^ undefined or not in scope
   |
   = help: you can import it into scope: `use syntax::symbol::InternedString;`.

error[E0412]: type name `token::InternedString` is undefined or not in scope
  --> /Users/myuser/.cargo/registry/src/github.com-1ecc6299db9ec823/stainless-0.1.9/src/test.rs:28:46
   |
28 |     pub fn failing_test(failing_msg: Option<(token::InternedString, ast::StrStyle)>) -> TestConfig {
   |                                              ^^^^^^^^^^^^^^^^^^^^^ undefined or not in scope
   |
   = help: you can import it into scope: `use syntax::symbol::InternedString;`.

error[E0433]: failed to resolve. Could not find `InternedString` in `parse::token`
  --> /Users/myuser/.cargo/registry/src/github.com-1ecc6299db9ec823/stainless-0.1.9/src/generate.rs:32:64
   |
32 |         let test_attribute = cx.attribute(sp, cx.meta_word(sp, token::InternedString::new("test")));
   |                                                                ^^^^^^^^^^^^^^^^^^^^^^^^^^ Could not find `InternedString` in `parse::token`

error[E0433]: failed to resolve. Could not find `InternedString` in `parse::token`
  --> /Users/myuser/.cargo/registry/src/github.com-1ecc6299db9ec823/stainless-0.1.9/src/generate.rs:35:62
   |
35 |         let should_panic = cx.attribute(sp, cx.meta_word(sp, token::InternedString::new("should_panic")));
   |                                                              ^^^^^^^^^^^^^^^^^^^^^^^^^^ Could not find `InternedString` in `parse::token`

error[E0433]: failed to resolve. Could not find `InternedString` in `parse::token`
  --> /Users/myuser/.cargo/registry/src/github.com-1ecc6299db9ec823/stainless-0.1.9/src/generate.rs:38:56
   |
38 |         let ignore = cx.attribute(sp, cx.meta_word(sp, token::InternedString::new("ignore")));
   |                                                        ^^^^^^^^^^^^^^^^^^^^^^^^^^ Could not find `InternedString` in `parse::token`

error[E0433]: failed to resolve. Could not find `InternedString` in `parse::token`
  --> /Users/myuser/.cargo/registry/src/github.com-1ecc6299db9ec823/stainless-0.1.9/src/generate.rs:40:57
   |
40 |         let non_snake_word = cx.meta_list_item_word(sp, token::InternedString::new("non_snake_case"));
   |                                                         ^^^^^^^^^^^^^^^^^^^^^^^^^^ Could not find `InternedString` in `parse::token`

error[E0433]: failed to resolve. Could not find `InternedString` in `parse::token`
  --> /Users/myuser/.cargo/registry/src/github.com-1ecc6299db9ec823/stainless-0.1.9/src/generate.rs:41:53
   |
41 |         let allow_non_snake_case = cx.meta_list(sp, token::InternedString::new("allow"),
   |                                                     ^^^^^^^^^^^^^^^^^^^^^^^^^^ Could not find `InternedString` in `parse::token`

error[E0433]: failed to resolve. Could not find `InternedString` in `parse::token`
  --> /Users/myuser/.cargo/registry/src/github.com-1ecc6299db9ec823/stainless-0.1.9/src/generate.rs:89:44
   |
89 |                     let should_panic_str = token::InternedString::new("should_panic");
   |                                            ^^^^^^^^^^^^^^^^^^^^^^^^^^ Could not find `InternedString` in `parse::token`

error[E0433]: failed to resolve. Could not find `InternedString` in `parse::token`
  --> /Users/myuser/.cargo/registry/src/github.com-1ecc6299db9ec823/stainless-0.1.9/src/generate.rs:90:40
   |
90 |                     let expected_str = token::InternedString::new("expected");
   |                                        ^^^^^^^^^^^^^^^^^^^^^^^^^^ Could not find `InternedString` in `parse::token`

error[E0433]: failed to resolve. Could not find `InternedString` in `parse::token`
   --> /Users/myuser/.cargo/registry/src/github.com-1ecc6299db9ec823/stainless-0.1.9/src/generate.rs:146:65
    |
146 |         let bench_attribute = cx.attribute(sp, cx.meta_word(sp, token::InternedString::new("bench")));
    |                                                                 ^^^^^^^^^^^^^^^^^^^^^^^^^^ Could not find `InternedString` in `parse::token`

error[E0425]: unresolved name `token::intern`
   --> /Users/myuser/.cargo/registry/src/github.com-1ecc6299db9ec823/stainless-0.1.9/src/lib.rs:188:9
    |
188 |         token::intern("describe"),
    |         ^^^^^^^^^^^^^ unresolved name

error: aborting due to 11 previous errors

error: Could not compile `stainless`.
```

Any ideas?

PS,
I couldn't find any blog post related to the error. So, i decided to ask you directlyblog post 

Thanks a lot

--- a/src/generate.rs
+++ b/src/generate.rs
@@ -9,7 +9,7 @@ use std::ops::Deref;
 use syntax::{ast, abi, codemap};
 use syntax::ptr::P;
 use syntax::ext::base;
-use syntax::parse::token;
+use syntax::symbol::Symbol;
 
 use syntax::ext::build::AstBuilder;
 
@@ -29,16 +29,16 @@ impl<'a> Generate<&'a DescribeState> for Test {
         let Test { description, block, test_config } = self;
 
         // Create the #[test] attribute.
-        let test_attribute = cx.attribute(sp, cx.meta_word(sp, token::InternedString::new("test")));
+        let test_attribute = cx.attribute(sp, cx.meta_word(sp, Symbol::intern("test")));
 
         // Create the #[should_panic] attribute.
-        let should_panic = cx.attribute(sp, cx.meta_word(sp, token::InternedString::new("should_panic")));
+        let should_panic = cx.attribute(sp, cx.meta_word(sp, Symbol::intern("should_panic")));
 
         // Create the #[ignore] attribute.
-        let ignore = cx.attribute(sp, cx.meta_word(sp, token::InternedString::new("ignore")));
+        let ignore = cx.attribute(sp, cx.meta_word(sp, Symbol::intern("ignore")));
 
-        let non_snake_word = cx.meta_list_item_word(sp, token::InternedString::new("non_snake_case"));
-        let allow_non_snake_case = cx.meta_list(sp, token::InternedString::new("allow"),
+        let non_snake_word = cx.meta_list_item_word(sp, Symbol::intern("non_snake_case"));
+        let allow_non_snake_case = cx.meta_list(sp, Symbol::intern("allow"),
                                                 vec![non_snake_word]);
         let allow_non_snake_case = cx.attribute(sp, allow_non_snake_case);
 
@@ -86,8 +86,8 @@ impl<'a> Generate<&'a DescribeState> for Test {
             match test_config.failing_msg {
                 Some(msg) => {
                     // Create #[should_panic(expected = "...")] attribute
-                    let should_panic_str = token::InternedString::new("should_panic");
-                    let expected_str = token::InternedString::new("expected");
+                    let should_panic_str = Symbol::intern("should_panic");
+                    let expected_str = Symbol::intern("expected");
                     let expected_name_value = cx.meta_name_value(
                         sp,
                         expected_str,
@@ -143,7 +143,7 @@ impl Generate<()> for Bench {
         let Bench { bench, description, block } = self;
 
         // Create the #[bench] attribute.
-        let bench_attribute = cx.attribute(sp, cx.meta_word(sp, token::InternedString::new("bench")));
+        let bench_attribute = cx.attribute(sp, cx.meta_word(sp, Symbol::intern("bench")));
 
         // Create the final Item that represents the benchmark.
         P(ast::Item {

--- a/src/lib.rs
+++ b/src/lib.rs
@@ -173,7 +173,8 @@ extern crate rustc_plugin;
 
 use self::describe::describe;
 use rustc_plugin as plugin;
-use syntax::parse::token;
+use syntax::symbol::Symbol;
+
 
 mod describe;
 mod parse;
@@ -185,7 +186,7 @@ mod generate;
 #[doc(hidden)]
 pub fn plugin_registrar(reg: &mut plugin::Registry) {
     reg.register_syntax_extension(
-        token::intern("describe"),
+        Symbol::intern("describe"),
         syntax::ext::base::IdentTT(Box::new(describe), None, false)
     );
 }

--- a/src/test.rs
+++ b/src/test.rs
@@ -6,7 +6,7 @@
 
 use syntax::ptr::P;
 use syntax::ast;
-use syntax::parse::token;
+use syntax::symbol::Symbol;
 
 /// A test as a description and associated block.
 #[derive(Clone)]
@@ -20,12 +20,12 @@ pub struct Test {
 pub struct TestConfig {
     pub ignored: bool,
     pub failing: bool,
-    pub failing_msg: Option<(token::InternedString, ast::StrStyle)>,
+    pub failing_msg: Option<(Symbol, ast::StrStyle)>,
 }
 
 impl TestConfig {
 
-    pub fn failing_test(failing_msg: Option<(token::InternedString, ast::StrStyle)>) -> TestConfig {
+    pub fn failing_test(failing_msg: Option<(Symbol, ast::StrStyle)>) -> TestConfig {
         TestConfig {
             failing: true,
             ignored: false,

